pkg/config/mesh/mesh.go

```diff
@@ -40,62 +40,143 @@ var (
+	if TestMode {
+		return meshconfig.MeshConfig{
+			IngressClass:                      "istio",
+			IngressControllerMode:             meshconfig.MeshConfig_STRICT,
+			ReportBatchMaxTime:                types.DurationProto(1 * time.Second),
+			ReportBatchMaxEntries:             100,
+			MixerCheckServer:                  "",
+			MixerReportServer:                 "",
+			DisablePolicyChecks:               true,
+			PolicyCheckFailOpen:               false,
+			SidecarToTelemetrySessionAffinity: false,
+			RootNamespace:                     constants.IstioSystemNamespace,
+			ProxyListenPort:                   15001,
+			ConnectTimeout:                    types.DurationProto(10 * time.Second),
+			IngressService:                    "istio-ingressgateway",
+			EnableTracing:                     true,
+			AccessLogFile:                     "/dev/stdout",
+			AccessLogEncoding:                 meshconfig.MeshConfig_TEXT,
+			DefaultConfig:                     &proxyConfig,
+			SdsUdsPath:                        "",
+			EnableSdsTokenMount:               false,
+			TrustDomain:                       "cluster.local",
+			TrustDomainAliases:                []string{},
+			Certificates:                      []*meshconfig.Certificate{},
+			DefaultServiceExportTo:            []string{"*"},
+			DefaultVirtualServiceExportTo:     []string{"*"},
+			DefaultDestinationRuleExportTo:    []string{"*"},
+			OutboundTrafficPolicy:             &meshconfig.MeshConfig_OutboundTrafficPolicy{Mode: meshconfig.MeshConfig_OutboundTrafficPolicy_ALLOW_ANY},
+			DnsRefreshRate:                    types.DurationProto(5 * time.Second), // 5 seconds is the default refresh rate used in Envoy
+			ProtocolDetectionTimeout:          types.DurationProto(100 * time.Millisecond),
+			EnableAutoMtls:                    &types.BoolValue{Value: false},
+			ThriftConfig:                      &meshconfig.MeshConfig_ThriftConfig{},
+			LocalityLbSetting:                 &v1alpha3.LocalityLoadBalancerSetting{},
+			ClusterLocalNamespaces:            append(make([]string, 0), defaultClusterLocalNamespaces...),
+		}
+		// Defaults matching the standard install
+		// order matches the generated mesh config.
+		return meshconfig.MeshConfig{
+			EnableTracing:               true,
+			AccessLogFile:               "",
+			AccessLogEncoding:           meshconfig.MeshConfig_TEXT,
+			AccessLogFormat:             "",
+			EnableEnvoyAccessLogService: false,
+			ReportBatchMaxEntries:       100,
+			ReportBatchMaxTime:          types.DurationProto(1 * time.Second),
+			DisableMixerHttpReports:     true,
+			DisablePolicyChecks:         true,
+			ProtocolDetectionTimeout:    types.DurationProto(100 * time.Millisecond),
+			IngressService:        "istio-ingressgateway",
+			IngressControllerMode: meshconfig.MeshConfig_STRICT,
+			IngressClass:          "istio",
+			TrustDomain:           "cluster.local",
+			TrustDomainAliases:    []string{},
+			SdsUdsPath:            "unix:/etc/istio/proxy/SDS",
```

Comment:
Nit: we can clean this up
```
pkg/test/framework/components/pilot/native.go
97:     // TODO(howardjohn) should we make this mesh wide default?
98-     m.SdsUdsPath = "unix:/etc/istio/proxy/SDS"
```

doesn't need to be this pr though

Reply:
I would think this should not even be exposed to user - certainly not a mesh default. 

A VM running as non-root may use a local path - we may allow some metadata to override the default path, but it should be per workload.

Reply:
I changed this to `./etc/...` in  https://github.com/istio/istio/pull/22679 by the way